musica-fortran/src/micm/micm_mod.F90

```diff
@@ -0,0 +1,63 @@
+module micm
+    use iso_c_binding
+
+    private
+    public :: micm_t
+
+    include "micm_c_def.F90"
+
+    type micm_t
+        private
+        type(c_ptr) :: ptr
+    contains
+        procedure :: delete => delete_micm
```

Comment:
@mattldawson is this a time when `finalize` could/should be used?

Reply:
yeah, I think so - if this function is always called right before the object is deleted, it could be:
```suggestion
        final :: delete_micm
```

Reply:
the function itself will have to be modified to take a `type(micm_t)` instead of a `class(micm_t)` with `intent(inout)`, but other than that should be mostly the same (I think). This `final` function gets called automatically when a `micm_t` object goes out of scope

Reply:
Would it be okay to leave the way it is then? It needs to be `class(micm_t)` for `micm_t` to have access to its members.

Reply:
I think you should still have access to the data members, you just would need to change it to:
```
    subroutine delete_micm(this)
        type(micm_t), intent(inout) :: this
        call delete_micm_c(this%ptr)
    end subroutine
```
I think the only difference between `class(foo_t)` and `type(foo_t)` is that `class(foo_t)` will accept types that extend `foo_t`, but `type(foo_t)` won't, and that type-bound procedures have to take the polymorphic `class` argument, but final routines need to take the concrete `type` argument.

Reply:
Ah, thank you for correcting me. I fixed it